File: components/src/dynamo/worker/myengine/args.py
# SPDX-FileCopyrightText: Copyright (c) 2025-2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
# SPDX-License-Identifier: Apache-2.0

"""Configuration and CLI argument parsing for the myengine sample backend."""

import argparse

from dynamo.common.backend import BackendCommonArgGroup, BackendCommonConfig


class Config(BackendCommonConfig):
    """MyEngine configuration. Inherits all common fields; no extras needed."""

    pass


def parse_args() -> Config:
    """Parse command-line arguments for the myengine backend."""
    parser = argparse.ArgumentParser(description="Dynamo myengine sample worker")
    BackendCommonArgGroup().add_arguments(parser)
    args = parser.parse_args()
    config = Config.from_cli_args(args)
    config.validate()
    return config

File: lib/sidecar/sglang/src/args.rs
// SPDX-FileCopyrightText: Copyright (c) 2025-2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

//! Command-line arguments for the SGLang sidecar.

use dynamo_sidecar_common::SidecarArgs;

/// Parsed sidecar arguments.
#[derive(clap::Parser, Debug, Clone)]
#[command(
    name = "dynamo-sglang-sidecar",
    about = "Dynamo sidecar for an out-of-process SGLang native gRPC server."
)]
pub struct Args {
    #[command(flatten)]
    pub sidecar: SidecarArgs,

    /// Maximum wait for a matching leader registration in telemetry mode.
    #[arg(long, default_value_t = 1800, value_parser = clap::value_parser!(u64).range(1..))]
    pub leader_discovery_timeout_secs: u64,

    /// Reachable host that decode workers use to connect to a prefill worker's
    /// SGLang disaggregation bootstrap port. By default this is derived from
    /// SGLang's concrete `host`, then `dist_init_addr`, then a routable local
    /// address. This is required when discovery exposes only loopback or
    /// wildcard addresses.
    #[arg(long, env = "SGLANG_DISAGGREGATION_BOOTSTRAP_HOST")]
    pub bootstrap_host: Option<String>,
}
